src/main/java/tars/Parser.java

```diff
@@ -0,0 +1,95 @@
+package tars;
+
+import java.util.ArrayList;
+/**
+ * Processes User inputs and executes commands based on inputs
+ * Has method to also list all Tasks added by User
+ *
+ * @author csk
+ * @version 1
+ */
+
+public class Parser {
+    private static final String LINE = "    _____________________________________________";
+
+    public Parser() {
+    }
+    /**
+     * Checks user inputs after being taken as string and string[] variables
+     * Chooses appropriate methods to call based on input given or throws error message
+     * depending on incorrect input received
+     *
+     * @param entryParts
+     * @param entry
+     * @param tasks
+     */
+    public String checkEntry(String[] entryParts, String entry, TaskList tasks) {
+        String result = "";
+        try {
+            if (entry.equals("todo") || entry.equals("deadLINE") || entry.equals("event")) {
+                throw new TarsException(LINE + "\n" + "    OOPS! Describe the task/event/deadLINE/todo or list"
+                        + "\n" + LINE);
+            } else if (entryParts[0].equals("list")) {
+                result = this.listPrint(tasks);
+            } else {
+                throw new TarsException(LINE + "\n" + "    OOPS! Only accept a task/event/deadLINE/todo as input"
+                        + "\n" + LINE);
+            }
+        } catch (TarsException e) {
+            result = e.getMessage();
+        }
+        return result;
+    }
+
+    /**
+     * Prints all the Tasks in the list at the current state of storage
+     *
+     * @param tasks
+     */
+    public String listPrint(TaskList tasks) {
+        String result = "";
+        StringBuilder strBuild = new StringBuilder();
+
+        if (tasks.getList().size() == 0) {
+            result = "No tasks added to list. Please add3 events/deadLINE/todos!";
+        } else {
+            for (int i = 0; i < tasks.getList().size(); i++) {
+                strBuild.append("    " + (i + 1) + ". " + tasks.getList().get(i) + "\n");
+            }
+            result = LINE + "\n" + "    Here are the tasks in your list:" + strBuild.toString() + "\n" + LINE;
+        }
+        return result;
+    }
+
+    /**
+     * Finds Task based on keyword stated by User, takes in entry of users and tasks in list
+     * Goes through all the tasks in the list that match keyword stated by User
```

Comment:
you could be more specific with the header comment, i.e. instead of using "goes through all the tasks" you could mention that it returns the task that keywords match
i know that you mean iterate through the list of tasks but it would be more concise to just describe what the method is supposed to do :D